### PR TITLE
wip(docs): refactor docs organization, intro tested docs

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR refactors the docs.steamship.com content as follows:

- Concepts
   - Agents
   - Packages
   - Plugins
   - Data Model
- Getting Started
   - Create Account
   - Your First Agent
- How To Guides
   - Use Agents and Tools
   - Import Data
   - Test and Debug 
   - ...
- Reference
   - Configuration
   - Python SDK
   - HTTP API

Notably, it proposes a paradigm for testing `How-To` code snippets (that _might_ produce more accurate docs).

It is very much a work-in-progress.

### CODE REVIEW

File: docs/getting-started/create-agent/create.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+To create your first agent, you'll need a GitHub account and a web browser.
+
+Our starter project is: https://github.com/steamship-core/multimodal-agent-starter
+
+To make your own agent using this repo, you may choose from the following options:
+
+1. Clone the repository locally
+1. Open the repository in a Github codespace.
+
+A third option of forking (or copying) the repository and then choosing one of the above two options exists as well.
```
Comment (dkolas) [inline docs/getting-started/create-agent/create.rst:19]: Yay

**Review by dkolas — COMMENTED** [review]
Docs updates look great.  I'm not sure I totally follow the strategy around including the python snippets, though?

File: docs/how-to/agents-and-tools/agent_with_search_tool.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+                # your agent's capabilities. The `SearchTool` uses Steamship's SERP API
+                # key for executing searches. There is no need to set up a separate account.
+                #
+                # If you have other tools you wish to use with your agent, you may add
+                # them here in a similar fashion.
+                SearchTool(),
+            ],
+            llm=ChatOpenAI(self.client, temperature=0),
+            conversation_memory=MessageWindowMessageSelector(k=2),
+        )
```
[the remark this answers is not in the document]
Comment (douglas-reid) [reply docs/how-to/agents-and-tools/agent_with_search_tool.py:23]: Lost in the madness of a billion changed files is: [tests/steamship_tests/docs/how-to/agents/tools/test_search.py](https://github.com/steamship-core/python-client/pull/472/files#diff-3bedd8f701915bf16beee5e715e4be3b3de18a160148b3545a60e55ee08d0a2e)

If you look closely, you can see that I pull the code into the test and deploy it and run it.

Open to other ideas, if you have any that make sense. I couldn't figure out how to get sphinx to pull files from outside of its root structure.

[the remark this answers is not in the document]
Comment (eob) [reply docs/how-to/agents-and-tools/agent_with_search_tool.py:23]: Mind blown. That is really cool

**Pull request closed without merging**